checker/src/org/checkerframework/checker/index/upperbound/UpperBoundAnnotatedTypeFactory.java

```diff
@@ -176,6 +197,74 @@ public void addComputedTypeAnnotations(Tree tree, AnnotatedTypeMirror type, bool
         }
     }
 
+    @Override
+    public void preProcessClassTree(ClassTree classTree) {
+        super.preProcessClassTree(classTree);
+        Element elt = TreeUtils.elementFromDeclaration(classTree);
+        List<? extends Element> enclosedElts =
+                ElementUtils.enclosingClass(elt).getEnclosedElements();
+        List<FlowExpressions.Receiver> rs = new ArrayList<>();
+        for (Element e : enclosedElts) {
+            if (!ElementUtils.isEffectivelyFinal(e)) {
```

Comment:
Why does it matter if the element is final?   The expression in annotation on the element may not be.